src/string.cr

```diff
@@ -2200,6 +2200,45 @@ class String
     match_result.try &.begin(0)
   end
 
+  # Searches sep or pattern (regexp) in the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
+  #
+  # ```
+  # "hello".rpartition("l") # => {"he", "l", "lo"}
+  # "hello".rpartition("x") # => {"hello", "", ""}
+  # ```
+  def partition(search : (Char | String)) : Tuple(String, String, String)
+    pre, mid, post = {"", "", ""}
```

Comment:
It's ok, in any case the performance should be the same. You can also do:

```cr
pre, mid, post = "", "", ""
```

That is, you don't need the tuple on the right. But the shortest one, maybe the clearest one, is the chained assignment.